factory/TriangleFactory.java

```diff
@@ -1,8 +1,23 @@
 package factory;
+import exceptions.NegativeLengthException;
+import exceptions.NotEnoughParamException;
+import shape.*;
+
 
 public class TriangleFactory implements ShapeFactory {
-  Shape create(String... params) {
-    // TODO: Do some parameter checking here and throw appropriate exceptions
-    return new Rect(Double.parseDouble(params[0]), Double.parseDouble(params[1]), Double.parseDouble(params[2]);
+  public Shape create(String... params) {
+	  int n = params.length;
+	  if (n == 3)
+	  {
+		  double a = Double.parseDouble(params[0]) ;
+		  double b = Double.parseDouble(params[1]) ;
+		  double c = Double.parseDouble(params[2]) ;
+		  if (a > 0 && c > 0 && b > 0 && a + b > c && a + c > b && c + b > a) 
+			  return new Triangle(a, b, c);
```

Comment:
I actually agree with your previous comment about checks in the shape classes.

Remember: the shape classes should have the business logic of the shapes, so they need to sanity check their input; however, the input from the user (making sure the string is a double etc) should be done in the factories. Its separation of responsibilities.